app/containers/ItunesProvider/ItunesDetailContainer/index.js

```diff
@@ -0,0 +1,66 @@
+import React, { memo } from 'react';
+import { selectSongs } from '../selectors';
+import { Collapse } from 'antd';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import { injectSaga } from 'redux-injectors';
+import styled from 'styled-components';
+import ItunesProviderSaga from '../saga';
+import PropTypes from 'prop-types';
+import { T } from '@app/components/T/index';
+
+const { Panel } = Collapse;
+
+const Wrapper = styled.div`
+  margin: auto;
+  padding: 20px 0px 20px;
+  max-width: 500px;
+`;
+
+const ItunesDetailContainer = ({ match, songs }) => {
+  const getSong = (id) => {
+    const foundSong = songs.results.find((song) => song.trackId === parseInt(id));
```

Comment:
iterating over all the entries in the redux store to get one value within the render is too computationally heavy. 


1. in the componentDidMount hook of the component, dispatch an action that contains the trackId
2. In the saga use the `select` hook from redux-saga, get all of the tracks that are in the store
3. search this list of tracks for the trackId that was passed in the action
4. If there is a match Dispatch the success action and update the `trackDetails` key in the redux store.
5. In case you don't find a track with a matching trackId in the redux store make an API call to the itunes API to fetch the track based on the trackId
6. Dispatch an action to update the list of tracks in the store and add the newly fetched item there. This will make sure that subsequent fetches for this trackId are fulfilled from the redux-store rather than by making an API call. 
7. This will also update a `trackDetails` key in the redux store.
8. Write a selector that reads the `trackDetails` from the reducer.
9. Add that to the mapStateToProps of the `ItunesDetailContainer`
10. Use that to render the details